apps/server/src/modules/tldraw/repo/tldraw.repo.ts

```diff
@@ -1,20 +1,59 @@
 import { EntityManager } from '@mikro-orm/mongodb';
 import { Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { BulkWriteResult, Collection, Sort } from 'mongodb';
+import { TldrawConfig } from '../config';
 import { TldrawDrawing } from '../entities';
 
 @Injectable()
 export class TldrawRepo {
-	constructor(private readonly _em: EntityManager) {}
+	private readonly collectionName: string;
 
-	async create(entity: TldrawDrawing): Promise<void> {
+	constructor(private readonly configService: ConfigService<TldrawConfig, true>, private readonly _em: EntityManager) {
+		this.collectionName = this.configService.get<string>('TLDRAW_DB_COLLECTION_NAME') ?? 'drawings';
+	}
+
+	public async create(entity: TldrawDrawing): Promise<void> {
 		await this._em.persistAndFlush(entity);
 	}
 
-	async findByDocName(docName: string): Promise<TldrawDrawing[]> {
+	public async findByDocName(docName: string): Promise<TldrawDrawing[]> {
 		return this._em.find(TldrawDrawing, { docName });
 	}
 
-	async delete(entity: TldrawDrawing | TldrawDrawing[]): Promise<void> {
+	public async delete(entity: TldrawDrawing | TldrawDrawing[]): Promise<void> {
 		await this._em.removeAndFlush(entity);
 	}
+
+	public get(query: object): Promise<TldrawDrawing | null> {
+		const collection = this.getCollection();
+		return collection.findOne<TldrawDrawing>(query, { allowDiskUse: true });
+	}
+
+	public async put(query: object, values: object): Promise<TldrawDrawing | null> {
+		const collection = this.getCollection();
+		await collection.updateOne(query, { $set: values }, { upsert: true });
+		return this.get(query);
+	}
+
+	public del(query: object): Promise<BulkWriteResult> {
+		const collection = this.getCollection();
+		const bulk = collection.initializeOrderedBulkOp();
+		bulk.find(query).delete();
+		return bulk.execute();
+	}
+
+	public readAsCursor(query: object, opts: { limit?: number; reverse?: boolean } = {}): Promise<TldrawDrawing[]> {
```

Comment:
We have types for sorting and pagination in shared. Can you please check if they work for you?

Reply:
If you mean the IFindOptions<T> then I don't think it really fits here.